osgp/protocol-adapter-mqtt/osgp-protocol-adapter-mqtt/src/main/java/org/opensmartgridplatform/adapter/protocol/mqtt/application/services/SubcriptionService.java

```diff
@@ -0,0 +1,129 @@
+            final MqttClientAdapterFactory mqttClientAdapterFactory,
+            final OutboundOsgpCoreRequestMessageSender outboundOsgpCoreRequestMessageSender,
+            @Value("#{new Integer('${mqtt.broker.defaultPort}')}") final int defaultPort,
+            @Value("${mqtt.broker.defaultTopics}") final String defaultTopics,
+            @Value("${mqtt.broker.defaultQos}") final String defaultQos) {
+        this.mqttDeviceRepository = mqttDeviceRepository;
+        this.mqttClientAdapterFactory = mqttClientAdapterFactory;
+        this.outboundOsgpCoreRequestMessageSender = outboundOsgpCoreRequestMessageSender;
+        this.defaultPort = defaultPort;
+        this.defaultTopics = defaultTopics;
+        this.defaultQos = defaultQos;
+    }
+
+    public void subscribe(final MessageMetadata messageMetadata) {
+        final MqttDevice device = this.getOrCreateDevice(messageMetadata);
+        final MqttClientAdapter mqttClientAdapter = this.mqttClientAdapterFactory.create(device, messageMetadata, this);
+        mqttClientAdapter.connect();
+    }
+
+    private MqttDevice getOrCreateDevice(final MessageMetadata messageMetadata) {
+        MqttDevice device = this.mqttDeviceRepository.findByDeviceIdentification(
+                messageMetadata.getDeviceIdentification());
+        if (device == null) {
+            device = new MqttDevice(messageMetadata.getDeviceIdentification());
+            device.setHost(messageMetadata.getIpAddress());
+            device.setPort(this.defaultPort);
+            device.setTopics(this.defaultTopics);
+            device.setQos(this.defaultQos);
+            this.mqttDeviceRepository.save(device);
+        }
+        return device;
+    }
+
+    @Override
+    public void onConnect(final MqttClientAdapter mqttClientAdapter, final Mqtt3ConnAck ack,
+            final Throwable throwable) {
+        if (throwable == null) {
+            this.onConnectSuccess(mqttClientAdapter, ack);
+        } else {
+            LOG.info(String.format("Client connect failed for device:%s",
+                    mqttClientAdapter.getMessageMetadata().getDeviceIdentification()), throwable);
+        }
+    }
+
+    private void onConnectSuccess(final MqttClientAdapter mqttClientAdapter, final Mqtt3ConnAck ack) {
+        LOG.info(String.format("Client connected for device:%s ack:%s",
+                mqttClientAdapter.getMessageMetadata().getDeviceIdentification(), ack.getType()));
+        final MqttDevice device = mqttClientAdapter.getDevice();
+        final MqttQos qos = this.getQosOrDefault(device);
+        final String[] topics = device.getTopics().split(",");
+        Arrays.stream(topics).forEach(topic -> mqttClientAdapter.subscribe(topic, qos));
+    }
+
+    private MqttQos getQosOrDefault(final MqttDevice device) {
+        MqttQos mqttQos;
+        try {
+            mqttQos = MqttQos.valueOf(device.getQos());
+        } catch (final IllegalArgumentException | NullPointerException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)

Reply:
Logged it